modules/core/src/main/java/org/apache/ignite/internal/processors/cache/binary/BinaryMetadataFileStore.java

```diff
@@ -255,6 +272,45 @@ void finishWrite(int typeId, int typeVer) {
         writer.finishWriteFuture(typeId, typeVer);
     }
 
+    /**
+     * Try looking for legacy directory with binary metadata and move it to new directory
+     */
+    private void fixLegacyFolder(String consistendId) throws IgniteCheckedException {
```

Comment:
I don't think we should keep this in production source code. This method used only once for the migration procedure. I think the right behaviour should be:
1. fail the node when users try to start a new Ignite version from the previous persistence directory.
2. provide migration scripts.